### PR TITLE
added asciidocs and java code for ehcache3, spring, jsr107 example #996

### PR DESCRIPTION
**Pull request by gibsong**

### CODE REVIEW

File: src/main/asciidoc/annotations.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+= JSR-107(JCache) Annotations
+
+If you are familiar with Spring, you know that it provides annotations to assist in developing applications.  In regards to caching, Spring offers support for two sets of annotations that can be used to implement caching.  You have the original Spring annotations and the new JSR-107 annotations.  The original Spring annotations are available to use with Spring versions 3.1+, while the JSR-107 annotations are only available in Spring 4.1+.  In this example we are going to use the JSR-107 annotations.  Below I have listed the most commonly used JSR-107(JCache) annotations, with brief descriptions as well as links to their API's.
+
+* *@CacheDefaults* - allows configuration of defaults at the class level.  For instance, you can define a cache name at the class level and this will be used as the default.
+** http://ignite.apache.org/jcache/1.0.0/javadoc/javax/cache/annotation/CacheDefaults.html
```
Comment (jhouserizer) [inline src/main/asciidoc/annotations.adoc]: I don't think we want to refer to (reference) apache ignite .   this is gemfire re-packaged as OSS and has become a competitor.  You should be able to find the official version of these here in github at the official jsr107 project.

https://github.com/jsr107/jsr107spec/tree/master/src/main/java/javax/cache/annotation

File: src/main/asciidoc/intro.adoc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+= Intro
+
+Ehcache 3.0.0 has been released!  In this post I would like to demonstrate how to use it as Spring's caching implementation.  This article will show you how to use Ehcache 3.0.0, Spring 4.2 and the JSR-107 annotations to build a simple Spring web application.  I have also included some background reading for those of you that would like to read more in depth on the subject matter.
```
Comment (jhouserizer) [inline src/main/asciidoc/intro.adoc]: 3.0.1 will be out in a couple of days.  Suggest you just say "Ehcache 3 has been released!"   and then "how to use Ehcache 3.0"  (drop the ".0".

File: src/main/asciidoc/intro.adoc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+= Intro
```
Comment (anthonydahanne) [inline src/main/asciidoc/intro.adoc]: I would add links to each project (ehcache3, spring, jsr107)

**Comment by jhouserizer** [comment]
Great!  - I didn't run the examples / try it, but I assume you did.  I think you did a great job with the article - and only put a few minor comments on the files in the commit.

File: src/main/asciidoc/steps.adoc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    <resources>
+      <heap unit="entries">2000</heap>
+      <offheap unit="MB">100</offheap> <!-- unit of measure is case sensitive! -->
+    </resources>
+  </cache-template>
+</config>
+----
++
+NOTE: XML Configuration Documentation: http://www.ehcache.org/documentation/3.0/xml.html
+. Create a cache by implementing the JCacheManagerCustomizer.customzie(CacheManager cacheManagegr) method, which will be invoked before the CacheManager is used.
```
Comment (anthonydahanne) [inline src/main/asciidoc/steps.adoc]: typo on customize
typo on cacheManager

File: src/main/asciidoc/annotations.adoc (hunk 1 above)
Comment (gibsong) [inline src/main/asciidoc/annotations.adoc]: Woops!  That was not my intention.  Good catch.  I just updated the urls.

File: src/main/asciidoc/intro.adoc (hunk 2 above)
Comment (gibsong) [inline src/main/asciidoc/intro.adoc]: I updated the into text to reflect you version naming.

File: src/main/asciidoc/intro.adoc (hunk 3 above)
Comment (gibsong) [inline src/main/asciidoc/intro.adoc]: done, there are now links to all three.  Please let me know if you don't think they link to the correct page?

File: src/main/asciidoc/steps.adoc (hunk 4 above)
Comment (gibsong) [inline src/main/asciidoc/steps.adoc]: fixed typos

**Pull request merged**